### ISSUE TITLE
Lack of application of discriminators in both dgmr.py and pseudocode/train.txt

### ISSUE THREAD

**Issue by rhanai** [openclimatefix/skillful_nowcasting#12]
Hi, I am also interested in the original paper and I'm glad to see recent great progress in this repo due to uploaded pseudocodes.

From what I read, I'm afraid that both pseudocode/train.txt (from DeepMind) and dgmr.py (in this repo) lack applying discriminators before calculating the generator loss.

* train.txt

```python:train.txt
def train():
  
  # ...

  gen_sequences = [tf.concat([batch_inputs, x], axis=1) for x in gen_samples]
  # HERE: loss_hinge_gen should accept Tensor containing discriminator scores instead of precipitation themselves.
  gen_disc_loss = loss_hinge_gen(tf.concat(gen_sequences, axis=0))
```

* dgmr.py
```
    def training_step(self, batch, batch_idx):

        # ..

        generated_sequence = [torch.cat([images, x], dim=2) for x in predictions]
        # HERE
        generator_disc_loss = loss_hinge_gen(torch.cat(generated_sequence, dim=0))
```

Because `loss_hinge_gen()` applies the whole-Tensor mean function inside, the resulting Tensor appears to have a correct shape.

Thanks.

**Comment by jacobbieker** [comment]
Thanks for pointing that out! Should have just pushed a fix!
